DEVELOPING.md

```diff
@@ -0,0 +1,53 @@
+
+Deserialization of Objects and their Properties (primitives that are read-from/written-to files) is mostly transparent to users/developers. Once you use the property macros (e.g. ` OpenSim_DECLARE_UNNAMED_PROPERTY`) in the header and perform the necessary wiring/calls, these properties know how to serialize themselves to XML elements and back. This serialization/deserialization, however, assumes a fixed XML layout/schema. If this assumption fails (because of a property name change, layout change or addition of new properties) then deserialization will not work out of the box and the developer making the change is responsible for changing the version number and the deserialization code to be in sync with the latest code. This is very important since if the developer does not update the version number then the code and the XML files will not be in sync and it will be very hard to go back to find out what changes should have been made. 
+
+Keep in mind that deserialization methods are practically called during the construction of the top level object/Model so typically you cannot rely on having any member variables or pointers populated unless done by default constructors, instead deserialization methods operate on the XML structure only. This helps isolate the backward compatibility code from the rest of the modeling/simulation or initialization functions.
+
+Deserialization works by making the following calls:
+ 1. A user constructs an object that is a subclass of `OpenSim::Object` by using a constructor that takes the file name of a .osim or .xml file (only some classes have such a constructor). The construction call sequence ends with calling `OpenSim::Object`'s constructor with the file name as an argument. This constructor invokes code that parses the XML file.
+ 2. The parsing code looks for XML tags that correspond to names of subclasses of `OpenSim::Object`; when it finds one of the them it performs the next 2 steps.
+ 3. An Object of the appropriate type is instantiated from the registry of available types using a lookup by the XML tag.
+ 4. The method `updateFromXMLNode(SimTK::Xml::Element& aNode, int versionNumber) ` is invoked on the object and is passed the XML element corresponding to the object along with the version number from the XML document/file. That’s exactly the hook to deserialization that developers can use to correct for deserialization changes by manipulating the passed in `aNode` to match what the latest code expects. 
```

Comment:
Please don't use `aNode` here (I'm begging!). The "a" prefix for arguments is leftover junk in the code and not part of the current coding standards. The argument name should be `node`. I know it might still have the old name in the code but we should fix that, and in user documentation not spread the ugliness any further.